crates/ruff/src/rules/flynt/rules/static_join_to_fstring.rs

```diff
@@ -44,21 +46,39 @@ fn build_fstring(joiner: &str, joinees: &[Expr]) -> Option<Expr> {
             fstring_elems.push(helpers::to_constant_string(joiner));
         }
         fstring_elems.push(helpers::to_fstring_elem(expr)?);
+
+        if let Expr::Constant(ast::ExprConstant {
+            value: Constant::Str(value),
+            ..
+        }) = expr
+        {
+            string_args.push(value);
+        }
     }
 
-    let node = ast::ExprJoinedStr {
-        values: fstring_elems,
-        range: TextRange::default(),
+    let node = if string_args.len() == joinees.len() {
+        ast::Expr::Constant(ast::ExprConstant {
+            value: Constant::Str(string_args.iter().join(joiner)),
+            range: TextRange::default(),
+            kind: None,
```

Comment:
`range` and `kind` are irrelevant since the autofix uses the range of the original `expr` (see the `Diagnostic`'s construction in `static_join_to_fstring`)